daemon/src/maker_inc_connections.rs

```diff
@@ -117,19 +118,54 @@ impl Actor {
         &mut self,
         mut stream: TcpStream,
         taker_address: SocketAddr,
-        ctx: &mut Context<Self>,
+        ctx: &mut xtra::Context<Self>,
     ) -> Result<()> {
         let transport_state = noise::responder_handshake(&mut stream, &self.noise_priv_key).await?;
         let taker_id = Identity::new(transport_state.get_remote_public_key()?);
 
-        tracing::info!(%taker_id, address = %taker_address, "New taker connected");
-
         let transport_state = Arc::new(Mutex::new(transport_state));
 
         let (read, write) = stream.into_split();
         let mut read =
             FramedRead::new(read, wire::EncryptedJsonCodec::new(transport_state.clone()));
-        let write = FramedWrite::new(write, EncryptedJsonCodec::new(transport_state));
+        let mut write = FramedWrite::new(write, EncryptedJsonCodec::new(transport_state));
+
+        match read
+            .try_next()
+            .timeout(Duration::from_secs(10))
+            .await
+            .with_context(|| {
+                format!(
+                    "Taker {} did not send Hello within 10 seconds, dropping connection",
+                    taker_id
+                )
+            })? {
+            Ok(Some(TakerToMaker::Hello(taker_version))) => {
+                let our_version = Version::current();
+                write.send(MakerToTaker::Hello(our_version.clone())).await?;
+
+                if our_version != taker_version {
+                    tracing::debug!(
+                        "Network version mismatch, we are on version {} but taker is on version {}",
+                        our_version,
+                        taker_version
+                    );
+
+                    // A taker running an old version is not treated as error for the maker
```

Comment:
Note: You are comparing `!=` but the comment says "older" which would imply a comparison like `<`?